dao/src/main/java/org/thingsboard/server/dao/sql/job/JobRepository.java

```diff
@@ -45,6 +45,8 @@ Page<JobEntity> findByTenantIdAndSearchText(@Param("tenantId") UUID tenantId,
     @Query("SELECT j FROM JobEntity j WHERE j.id = :id")
     JobEntity findByIdForUpdate(UUID id);
 
+    JobEntity findByTenantIdAndKey(@Param("tenantId") UUID tenantId, @Param("key") String key);
+
     boolean existsByKeyAndStatusIn(String key, List<JobStatus> statuses);
```

Comment:
I know it's not your method, by please refactor it to include tenant id in the query